### PR TITLE
[#1524] made InjectionExtension compatible with older junit5 versions again

### PR DESCRIPTION
**Pull request by cdietrich**
[#1524] made InjectionExtension compatible with older junit5 versions again

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java [hunk 1]
```diff
@@ -59,10 +59,14 @@ public void beforeEach(ExtensionContext context) throws Exception {
...
 				injector.injectMembers(testInstance);
-				for (Object o : requiredTestInstances.getEnclosingInstances()) {
-					injector.injectMembers(o);
+				try {
+					TestInstances requiredTestInstances = context.getRequiredTestInstances();
+					for (Object o : requiredTestInstances.getEnclosingInstances()) {
+						injector.injectMembers(o);
+					}
+				} catch (NoSuchMethodError e) {
+					// OK, getRequiredTestInstances is not there in Junit5 < 5.4
```
Comment (szarnekow) [inline org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java:69]: Should we throw an `ExtensionConfigurationException` if the `testInstance` is an instance of a non-static inner class?

Comment (cdietrich) [reply org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java:69]: ähm. this is all about non static inner classes

Comment (szarnekow) [reply org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java:69]: I mean from within the catch block.

Comment (cdietrich) [reply org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java:76]: updated the PR

**Comment by cdietrich** [comment]
@frankbenoit  can you have a quick look if this still fits your needs?

**Comment by frankbenoit** [comment]
:+1:

**Pull request merged**